src/main/java/edu/harvard/iq/dataverse/engine/command/impl/CreateDatasetCommand.java

```diff
@@ -136,15 +136,16 @@ public Dataset execute(CommandContext ctxt) throws CommandException {
                 && theDataset.getGlobalIdCreateTime() == null) {
             if (protocol.equals("doi")) {
                 String doiRetString = "";
+                IdServiceBean idServiceBean = IdServiceBean.getBean(ctxt);
                 try{
                     logger.log(Level.FINE,"creating identifier");
-                    doiRetString = IdServiceBean.getBean(ctxt).createIdentifier(theDataset);
+                    doiRetString = idServiceBean.createIdentifier(theDataset);
                 } catch (Exception e){
                     logger.log(Level.WARNING, "Exception while creating Identifier: " + e.getMessage(), e);
                 }
 
                 // Check return value to make sure registration succeeded
-                if (doiProvider.equals("EZID") && doiRetString.contains(theDataset.getIdentifier())) {
+                if (!idServiceBean.registerWhenPublished() && doiRetString.contains(theDataset.getIdentifier())) {
```

Comment:
It's my intention to drop the check on the `protocol` once implementation of the handle service bean is completed. 

There is still `DestroyDatasetCommand` calling `ctxt.doiDataCite()`. The last DatasetCommand that is not agnostic about the configured bean by calling `IdServiceBean.getBean(ctxt)` controlled with another property of an `idServiceBean`. Don't yet know how that property should be named and what the value for both properties should be for handlenet. Even I'm not sure whether these properties should be hard-coded or configurable. I also remember Ben wanted a handle to point to some resource with some original meta data and additional provenance about why/when the dataset was destroyed and who requested/executed/approved the action.

Rediscovering the [API](https://docs.google.com/document/d/1KW_zticHqudDjeVD7UjFq9OlnY5eJAqpGL87tVegwSc/) I guess the interface should be renamed into `IdRegistrationServiceBean`.